### DISCUSSION TITLE
Fostering strong communities and collaboration between companies for corporate sponsored open sourced packages that are business critical

### DISCUSSION THREAD

**Discussion by epijim** [rinpharma/positconf-roundtables-2023#5] [Ideas]
Open sourcing is the first step on a journey to meaningful collaboration. 

Where are we today? Where have we seen collaboration and aligned roadmaps? Are there gaps where we could be collaborating more? How could we better capture and align where we could be collaborating across the spectrum of drug development from research, clinical reporting and access?

**Comment by MichaelRimler** [comment]
I like the theme of Collaborations.

**Comment by MikeKSmith-Pfizer** [comment]
Fostering strong communities and collaboration is vital to drive things forward in this space. I hope what we have goes beyond a collective of developers who happen to be working in the same domain. For example, tidyverse packages have a strong design philosophy that helps them play well together. Same with the tidymodels-verse. One question that senior managers may ask is "How can we trust externally developed packages?". So it might be good to discuss how this group / pharmaverse can build trust. 

(There's still a lot of fear, uncertainty, doubt in senior management, at least in my experience. Even though we've been talking about what open-source development means for over 15 years).

**Reply by MichaelRimler** [reply]
Another related topic (which Joe Rickert has mentioned to me) is about the long term sustainability of packages currently being released (especially those primarily driven be a few people and are being integrated into data reporting workflows). How do we manage the risk of people 'moving on' to other activities (due to loss of interest, lack of resource, retirement)? And who is the 'we' in that question? 

How has this been managed elsewhere?

**Reply by dgkf-roche** [reply]
I filed a related discussion in #15, thinking about how we build expectations for the use of code (and if need be, the _misuse_ of code). When I look at other cross-industry projects, I see things like NumFOCUS, Apache, etc. Are we missing something by not having that sort of managing entity?

**Reply by kamilsi** [reply]
@MikeKSmith-Pfizer's observations about managerial concerns with open-source resonate with me. I'm in a unique position that allows me to bridge the worlds of management and development. I grasp the enthusiasm developers have for open sourcing, but I also comprehend the reservations held by managers. In my conversations with fellow managers, the key to overcoming these reservations has been to translate the developers' passion into the language of business objectives. A well-defined plan that aligns our open-source initiatives with these objectives has been instrumental in alleviating managerial concerns.

We've successfully integrated an R-based API into our Java-based eCRF platform, a move that has not only reduced costs but also encouraged collaboration. This project even drew valuable contributions from a client. However, the journey has been far from smooth. Partners and clients frequently raise concerns about "blurred responsibilities," particularly as our proprietary software begins to transform into a co-developed, open-source project. Questions like, "Who is responsible if something goes wrong?" inevitably come up, along with the emergence of new kinds of long-term responsibilities that were not part of the project's original scope—increasing both risks and their duration.

To address these challenges, we're taking proactive steps to formalize roles, create comprehensive planning frameworks, and promote transparent communication among all involved parties. While we have our own approach and I remain optimistic about our journey, I'm keenly interested in learning from the community. There may well be better practices out there that we could adopt to improve our open-source endeavors in the pharma sector. Looking forward to engaging with you all on these topics. See you there!
